lib/azure/armrest/configuration.rb

```diff
@@ -162,7 +162,7 @@ def token_expiration
 
       # Return the default api version for the given provider and service
       def provider_default_api_version(provider, service)
-        @provider_api_versions[provider.downcase][service.downcase]
+        @provider_api_versions[provider.downcase][service.downcase] rescue nil
       end
```

Comment:
```rescue nil``` is going to rescue from any exception. If this rescue statement is designed to handle the situation where the provider and service keys are not present then ```try``` method might be better option.